components/constellation/constellation.rs

```diff
@@ -2238,7 +2235,22 @@ where
                 let send = match browsing_context_group.webgpus.entry(host) {
                     Entry::Vacant(v) => v
                         .insert(match WebGPU::new() {
-                            Some(webgpu) => webgpu,
+                            Some(webgpu) => {
+                                let msg = ConstellationControlMsg::SetWebGPUPort(webgpu.1);
+                                match source_pipeline {
```

Comment:
given the previous match on the `source_pipeline`, it looks like it has to be Some() here.
I think we should refactor it to extract it right away in the beginning:
```rust
let source_pipeline = match self.pipelines.get(&source_pipeline_id) {
  Some(pipeline) => pipeline,
  None => return warn!(...),
};
```